coordinator/tests/integration/helpers.py

```diff
@@ -364,3 +414,37 @@ def get_ingress_proxied_hostname(juju: Juju):
             "proxied-endpoints"
         ]
     )[TRAEFIK_APP]["url"].split("://")[1]
+
+
+def service_mesh(
+    enable: bool,
```

Comment:
I think it would make sense to make this a contextmanager, and use a nullcontext for the 'mesh disabled' case.

```suggestion
@contextmanager
def service_mesh(
```

Reply:
`service_mesh` is now a context manager